### ISSUE TITLE
Accessibility:Slider

### ISSUE THREAD

**Issue by Sanjay-1312** [SAP/fundamental-ngx#5081]
Accessibility: Slider
 (There are couple of accessibility issues found while testing, so i have attached error document for you reference.)
Screenshot:
[Slider_Core_ARIBA Controls.docx](https://github.com/SAP/fundamental-ngx/files/6262546/Slider_Core_ARIBA.Controls.docx)

Application Name:	ARIBA Controls- Fundamental Library for Angular
URL:	https://fundamental-ngx.netlify.app/#/platform/home
Web browser:	Google Chrome 89.0.4389.82
Operating System:	Microsoft Windows 10 x64
Screen Reader:	JAWS 2021.2012.57.400

### Issue 1

Priority:	High
Checkpoint:	ACC-264
(264.1)

Press tab to navigate to any slider available.
Observed Behavior:
On navigating to the sliders available, information regarding the ‘maximum and minimum value’ available is not read. Therefore, the information regarding the maximum and minimum value is not conveyed to the screen reader user.
The speech output is:
slider left right slider  Has Popup  
50
To increase or decrease use the Arrow keys.
slider left right slider  Has Popup  
0
To increase or decrease use the Arrow keys.
Expected Behavior:
Ideally, on navigating to the slider using tab key, information regarding the ‘maximum and minimum value’ available should be read. Therefore, the information read should be clear and precise.
Screenshot:

![image](https://user-images.githubusercontent.com/52071505/122223300-ad3e6900-ced0-11eb-9654-5a5fce5115db.png)

### Issue 2

Priority:	Medium
Checkpoint:	ACC-264 (264.1)

Press tab to navigate to any slider.
Observed Behavior:
On navigating to the sliders, an additional information ‘has pop up’ is read. Also, information ‘slider’ is read twice. Therefore, additional information is read on navigation.
The speech output is:
slider left right slider  Has Popup  
50
To increase or decrease use the Arrow keys.
slider left right slider  Has Popup  
0
To increase or decrease use the Arrow keys.
Expected Behavior:
Ideally, no additional information should be read on navigation and the information read should be clear and precise.
Screenshot:

![image](https://user-images.githubusercontent.com/52071505/122223543-e545ac00-ced0-11eb-978c-22b036a9f3e6.png)

### Issue 3

Priority:	High
Checkpoint:	ACC-270 (270.1)

Within ‘custom’ header, navigate to the slider and navigate using arrow keys
Observed Behavior:
The value within the slider cannot be changed using keyboard. i.e. the value cannot be changed using arrow keys. Therefore, the slider is not accessible with keyboard.
The speech output is:
slider left right slider  Has Popup  
0
To increase or decrease use the Arrow keys.
Expected Behavior:
Ideally, all the UI elements should be accessible using keyboard. i.e. the user should be able to change the value within the slider using arrow keys as mentioned within the tutor message.
Screenshot:
**Case 1**

![image](https://user-images.githubusercontent.com/52071505/122223684-07d7c500-ced1-11eb-8c3c-36cd816ca59e.png)

**Case 2 - Range Slider**
![image](https://user-images.githubusercontent.com/52071505/122223764-1c1bc200-ced1-11eb-80ed-5bf3b17f3acd.png)

### Issue 4

Priority:	Medium
Checkpoint:	ACC-261(261.2)

Observed Behavior:
No sufficient color contrast ratio of 4.5:1, highlighted in red.
Expected Behavior:
Ideally, all the element should meet the color contrast ratio of 4.5:1.
Screenshot:
**Case 1**
![image](https://user-images.githubusercontent.com/52071505/122223897-3a81bd80-ced1-11eb-9f31-cb5c52eb9df7.png)

**Case 2**
![image](https://user-images.githubusercontent.com/52071505/122223967-49687000-ced1-11eb-8100-a8fc231c8698.png)

**Case 3**

![image](https://user-images.githubusercontent.com/52071505/122224022-55543200-ced1-11eb-965b-f365d6fb09bc.png)

**Case 4**

![image](https://user-images.githubusercontent.com/52071505/122224097-669d3e80-ced1-11eb-8d18-c510a46cbfcc.png)

### Issue 5

Priority:	High
Checkpoint:	ACC-255
(255.1)

Observed Behavior:
The ToolTip or Text Alternative is not provided for the Slider, highlighted red.
Expected Behavior:
Ideally, the text alternative or tool tip should be provided to the icons UI elements to identify the user easily.
Note: For all the slider in the Slider Control doesn’t have the Text Alternative or Tool Tip for the user.
Screenshot:
![image](https://user-images.githubusercontent.com/52071505/122224209-83d20d00-ced1-11eb-8f48-aeb0cc0a04f6.png)

### Issue 6

Priority:	High
Checkpoint:	ACC-264
(264.1)

Observed Behavior:
The association label and Values is not read for the user, highlighted in red.
Expected Behavior:
Ideally, the visible content of the screen must be conveyed to the screen reader user.
Note: For all the input field and Combo box in the Slider Control doesn’t convey the association label for the user.
Screenshot:
Case 1:

![image](https://user-images.githubusercontent.com/52071505/122224336-a532f900-ced1-11eb-9980-a7c172e4d838.png)

Case 2:

![image](https://user-images.githubusercontent.com/52071505/122224392-b0862480-ced1-11eb-8612-72281ddf4466.png)

remove

### Issue 7

Priority:	High
Checkpoint:	ACC-264
(264.1)

Press F12 key and Bring the screen to 200% Zoom.
Adapt the screen to 320 Pixels.
Observed Behavior:
On adapting the screen to 200% Zoom and Responsive design, the horizontal bar is available.
Expected Behavior:
Ideally, the horizontal scroll bar should be removed.
**Note: The issue persists throughout all ‘Slider’.**
Screenshot:
Case 1:

![image](https://user-images.githubusercontent.com/52071505/122224676-f93ddd80-ced1-11eb-80ab-417c79a5d6ac.png)

Case 2:

![image](https://user-images.githubusercontent.com/52071505/122224732-06f36300-ced2-11eb-861e-ff3b146fc480.png)

**Comment by DeepakSap14** [comment]
@puru-hk 
please raise Issue 4 in fundamental-styles.
do not include playground, in the issue. That is not related to slider component.

**Comment by Sanjay-1312** [comment]
Issue 1: Fixed

Issue 2: 
![image](https://user-images.githubusercontent.com/78714518/124892921-ecf00080-dff7-11eb-902c-5dd1cde6c947.png)
OB: Throughout the control, on changing the values read thrice to the user.
EB: On changing the value using up/down arrow keys the changed value should be read once to the user.
Note: throughout the control

Issue 3: 
![image](https://user-images.githubusercontent.com/78714518/124892225-502d6300-dff7-11eb-9600-949c01856430.png)
OB: On accessing the slider as visually there is no pop up read as has pop up.
EB: ideally, such additional information such as 'Has pop up' should be removed.

Issue 4: Fixed

Issue 5: Fixed

Issue 6: Fixed

Issue 7: 
![image](https://user-images.githubusercontent.com/78714518/124892882-e2356b80-dff7-11eb-8856-4ab15f57aabd.png)
OB: on adapting the screen to320 CSS, the text gets truncated.
EB: Ideally, the texts shouldn't get truncated when screen is resized to 320 pixels. It should be clearly visible.

**Comment by Sanjay-1312** [comment]
Hello Colleagues,

issue 2:
![image](https://user-images.githubusercontent.com/78714518/125588330-99ccafcc-1add-48b7-a95d-6694ed67d55f.png)

OB: On changing the value using up/down arrow keys, the value is being read to the user 
EB: On changing the value using up/down arrow keys the changed value should be read once to the user.

Issue 3:
Same issue persists.
OB: an additional information such has slider has pop up is being read to the user.
EB: Ideally, such additional information should not be read to the user.

issue 7:
Same issue persists.
The dates are truncated.

URL:  https://deploy-preview-5683--fundamental-ngx.netlify.app/#/core/slider
JAWS 2021

**Comment by Sanjay-1312** [comment]
On increasing and decreasing the changed values are read twice using up and down arrow key.
Its known JAWS issue.
